elements/rh-scheme-toggle/test/rh-scheme-toggle.spec.ts

```diff
@@ -0,0 +1,114 @@
+
+function press(key: string) {
+  return async function() {
+    await sendKeys({ press: key });
+  };
+}
+
+describe('<rh-scheme-toggle>', function() {
+  describe('simply instantiating', function() {
+    let element: RhSchemeToggle;
+
+    beforeEach(async function() {
+      element = await createFixture<RhSchemeToggle>(
+        html`<rh-scheme-toggle></rh-scheme-toggle>`
+      );
+      await element.updateComplete;
+    });
+
+    it('imperatively instantiates', function() {
+      expect(document.createElement('rh-scheme-toggle')).to.be.an.instanceof(
+        RhSchemeToggle
+      );
+    });
+
+    it('should upgrade', async function() {
+      const klass = customElements.get('rh-scheme-toggle');
+      expect(element)
+          .to.be.an.instanceOf(klass)
+          .and.to.be.an.instanceOf(RhSchemeToggle);
+    });
+
+    it('should be accessible', async function() {
+      await expect(element).to.be.accessible();
+    });
+  });
+  describe('keyboard navigation', function() {
+    let element: RhSchemeToggle;
+
+    beforeEach(async function() {
+      element = await createFixture<RhSchemeToggle>(
+        html`<rh-scheme-toggle></rh-scheme-toggle>`
+      );
+    });
+    beforeEach(async () => await element.updateComplete);
+
+    describe('first tab press', function() {
+      beforeEach(press('Tab'));
+
+      it('should focus and check the device default mode radio input', async function() {
+        const snapshot = await a11ySnapshot();
+        expect(snapshot).to.have.axQuery({ name: 'Device default', focused: true, checked: true });
+      });
+
+      describe('right arrow press', function() {
+        beforeEach(press('ArrowRight'));
+
+        it('should focus and check the light mode radio input', async function() {
+          const snapshot = await a11ySnapshot();
+          expect(snapshot).to.have.axQuery({ name: 'Light', focused: true, checked: true });
+        });
+      });
+
+      describe('left arrow press', function() {
+        beforeEach(press('ArrowRight'));
+
+        it('should focus and check the dark mode radio input', async function() {
+          const snapshot = await a11ySnapshot();
+          expect(snapshot).to.have.axQuery({ name: 'Dark', focused: true, checked: true });
+        });
+      });
+    });
+
+    describe('local storage', function() {
+      beforeEach(function() {
+        localStorage.clear();
+      });
+
+      it('if local storage is set to light dark, should focus, check and set the scheme to light dark', async function() {
+        localStorage.setItem('rhdsColorScheme', 'light dark');
+        // Create new element after setting localStorage so it reads the value during initialization
+        const newElement = await createFixture<RhSchemeToggle>(
+          html`<rh-scheme-toggle></rh-scheme-toggle>`
+        );
+        await newElement.updateComplete;
+        expect(newElement.scheme).to.equal('light dark');
+      });
+
```

Comment:
prefer to do actions/side effects in describe/beforeEach. `it` should only relate to outcomes

```suggestion
    describe('with empty localStorage', function() {
      beforeEach(function() {
        localStorage.clear();
      });
      
      describe('with stored scheme set to `light dark`', function() {
        beforeEach(async function() {
          localStorage.setItem('rhdsColorScheme', 'light dark');
        });
        
        describe('adding basic toggle element', function() {
          let element: RhSchemeToggle;
          beforeEach(async function() {
            element = await createFixture<RhSchemeToggle>(html`<rh-scheme-toggle></rh-scheme-toggle>`);
          });
          it('uses the stored scheme', async function() {
            expect(newElement.scheme).to.equal('light dark');
          });
        });
      });
      
```

etc